pkg/source/algebrav1/queries.go

```diff
@@ -0,0 +1,120 @@
+package algebrav1
+
+import (
+	"bytes"
+	"fmt"
+	"math/big"
+	"strings"
+	"text/template"
+)
+
+type PoolsListQueryParams struct {
+	AllowSubgraphError     bool
+	LastCreatedAtTimestamp *big.Int
+	First                  int
+	Skip                   int
+	LastPoolIdsQuery       string
+}
+
+type PoolTicksQueryParams struct {
+	AllowSubgraphError bool
+	PoolAddress        string
+	Skip               int
+}
+
+func getPoolsListQuery(allowSubgraphError bool, lastCreatedAtTimestamp *big.Int, lastPoolIds []string, first, skip int) string {
+	var tpl bytes.Buffer
+	var lastPoolIdsQ string
+	if len(lastPoolIds) > 0 {
+		lastPoolIdsQ = fmt.Sprintf(", id_not_in: [\"%s\"]", strings.Join(lastPoolIds, "\",\""))
+	} else {
+		lastPoolIdsQ = ""
+	}
+	td := PoolsListQueryParams{
+		allowSubgraphError,
+		lastCreatedAtTimestamp,
+		first,
+		skip,
+		lastPoolIdsQ,
+	}
+
+	// Add subgraphError: allow
+	t, err := template.New("poolsListQuery").Parse(`{
+		pools(
+			{{ if .AllowSubgraphError }}subgraphError: allow,{{ end }}
+			where: {
+				createdAtTimestamp_gte: {{ .LastCreatedAtTimestamp }}
+				{{ .LastPoolIdsQuery }}
```

Comment:
Does it should be `.LastPoolIdsQ` instead?
```suggestion
				{{ .LastPoolIdsQ }}
```